storage/innobase/srv/srv0srv.cc

```diff
@@ -2881,6 +2883,8 @@ DECLARE_THREAD(srv_master_thread)(
 		}
 
 		srv_enable_undo_encryption_if_set();
+
+		log_check_new_key_version();
```

Comment:
Ok, master thread is used for this. For regular tabespace, to do this version checking, encryption threads are used right?

Why is this not part of encryption threads?

Reply:
Because encryption threads do more and differently: those also reencrypt earlier records with the new key.

With redo, that's not needed, and this way it works without encryption threads on.